docs/index.md

```diff
@@ -45,3 +45,156 @@ or the WCS API (which will also install `wcs-lib`):
 opam install wcs-api
 ```
 
+# Example
+
+In order to illustrate the use of wcs-ocaml, we are going to program a
+bot that tells a knock knock joke.
+
+Let's start to create a dialog node that says `"Knock knock"`:
+
+```ocaml
+let knock =
+  Wcs.dialog_node "Knock"
+    ~conditions: "true"
+    ~text: "Knock knock"
+	()
+```
+
+The function
+[`Wcs.dialog_node`](https://ibm.github.io/wcs-ocaml/wcs-lib/Wcs/index.html#val-dialog_node)
+creates a value of type
+[`Wcs_t.dialog_node`](https://ibm.github.io/wcs-ocaml/wcs-lib/Wcs_t/index.html#type-dialog_node)
+that corresponds to a JSON object of type
+[`DialogNode`](https://www.ibm.com/watson/developercloud/conversation/api/v1/)
+in WCS.
+
+The user is expected to ask _who is there?_. To capture this intent
+but not match exactly on this input, we can define a WCS intent with some examples to train the Natural Language Understanding component of WCS:
+
+```ocaml
+let who_intent =
+  Wcs.intent "Who"
+    ~examples: [
+      "Who's there?";
+      "Who is there?";
+      "Who are you?";
+    ]
+    ()
+```
+
+So, we can know define the next node in the conversation that is going
+to respond to the _who is there?_ question:
+
+```ocaml
+let whoisthere =
+  Wcs.dialog_node "WhoIsThere"
+    ~conditions_spel: (Spel.intent who_intent)
+    ~text: "Broken Pencil"
+    ~parent: knock
+    ()
+```
+
+We can notice that here the condition is not given as a string but as
+an expression written using the embedding of the Spel expression
+language (used by WCS) inside OCaml.
+
+In the rest of the conversation the user is expected to repeat the
+name of the character given by the bot. To be sure that the user
+repeat the name, we are going to define an entity `char_entity`
+containing the name or some synonyms:
```

Comment:
We now expect the user to repeat the name of the character mentioned by the bot.
To test that the user input matches the same character, we define an entity `char_entity` containing the name and a list of synonyms.